internal/testrunner/runners/pipeline/runner_test.go

```diff
@@ -318,3 +320,59 @@ func TestDiffUlite(t *testing.T) {
 		})
 	}
 }
+
+var jsonUnmarshalUsingNumberTests = []struct {
+	name string
+	msg  string
+	want interface{}
+}{
+	{
+		name: "empty",
+		msg:  "", // Will error "unexpected end of JSON input".
+	},
+	{
+		name: "string",
+		msg:  `"message"`,
+		want: "message",
+	},
+	{
+		name: "long",
+		msg:  "9223372036854773807",
+		want: json.Number("9223372036854773807"),
+	},
+	{
+		name: "array",
+		msg:  "[1,2,3,4,5]",
+		want: []interface{}{json.Number("1"), json.Number("2"), json.Number("3"), json.Number("4"), json.Number("5")},
+	},
+	{
+		name: "object",
+		msg:  `{"key":42}`,
+		want: map[string]interface{}{"key": json.Number("42")},
+	},
+	{
+		name: "object",
+		msg:  `{"key":42}answer`, // Will error "invalid character 'a' after top-level value".
+	},
+}
+
+func TestJsonUnmarshalUsingNumberTests(t *testing.T) {
+	for _, test := range jsonUnmarshalUsingNumberTests {
+		t.Run(test.name, func(t *testing.T) {
+			var got interface{}
+			err := jsonUnmarshalUsingNumber([]byte(test.msg), &got)
+			jerr := json.Unmarshal([]byte(test.msg), new(interface{}))
+
+			// String comparison because we are not returning a real json.SyntaxError.
+			if fmt.Sprint(err) != fmt.Sprint(jerr) {
+				t.Errorf("unexpected error: got:%#v want:%#v", err, jerr)
+			}
+			if err != nil {
+				return
+			}
+			if !reflect.DeepEqual(got, test.want) {
```

Comment:
Hmm.. Isn't there any replacement for `assert.DeepEqual` in `testify/assert`, so we need to use reflection here?

Reply:
`assert.Equal` calls `reflect.DeepEqual` in [a round about way that](https://github.com/stretchr/testify/blob/acba37e5db06f0093b465a7d47822bf13644b66c/assert/assertions.go#L58-L76). I try to avoid testing frameworks as much as possible unless they add value.

Reply:
Yes, in this case, it's about consistency with the rest of the codebase, so kindly please adjust to it.